src/test/java/TasksServletTest.java

```diff
@@ -205,12 +209,42 @@ public void allTasks() throws Exception {
     Mockito.when(tasksClient.listTaskLists()).thenReturn(ONE_TASKLIST);
     Mockito.when(tasksClient.listTasks(TASKLIST_ONE)).thenReturn(ALL_TASKS);
     servlet.doGet(request, response);
-
     TasksResponse tasksResponse = gson.fromJson(stringWriter.toString(), TasksResponse.class);
     Assert.assertEquals(ONE_TASKLIST_TITLES, tasksResponse.getTaskListNames());
     Assert.assertEquals(3, tasksResponse.getTasksToComplete());
     Assert.assertEquals(1, tasksResponse.getTasksDueToday());
     Assert.assertEquals(1, tasksResponse.getTasksCompletedToday());
     Assert.assertEquals(1, tasksResponse.getTasksOverdue());
   }
+
+  @Test
+  public void postTaskNullTaskListId() throws Exception {
+    servlet.doPost(request, response);
+    Assert.assertEquals(400, response.getStatus());
+  }
+
+  @Test
+  public void postTaskEmptyTaskListId() throws Exception {
+    Mockito.when(request.getParameter("taskListId")).thenReturn("");
+    servlet.doPost(request, response);
+    Assert.assertEquals(400, response.getStatus());
+  }
+
+  @Test
+  public void postValidTask() throws Exception {
+    Mockito.when(request.getParameter("taskListId")).thenReturn(TASKLIST_ID_ONE);
+    Mockito.when(tasksClient.postTask(TASKLIST_ID_ONE, validTask)).thenReturn(validTask);
+
+    StringReader reader = new StringReader(VALID_TASK_JSON);
+    BufferedReader bufferedReader = new BufferedReader(reader);
+    Mockito.when(request.getReader()).thenReturn(bufferedReader);
+
+    servlet.doPost(request, response);
+
+    Task postedTask = gson.fromJson(stringWriter.toString(), Task.class);
+
+    Assert.assertEquals(validTask.getTitle(), postedTask.getTitle());
+    Assert.assertEquals(validTask.getNotes(), postedTask.getNotes());
+    Assert.assertEquals(validTask.getDue(), postedTask.getDue());
```

Comment:
Propagating Anton's feedback on my TasksResponse tests - you can override equals in Task to compare these two directly.

Reply:
Task is a final, external class - I'm not sure how I would override this, or if it's a good idea to try (generally overriding external dependencies is bad practice AFAIK)

Reply:
Right - my bad. I was assuming Task was an internal class. Thanks for pointing this out!